usr/src/uts/common/fs/nfs/nfs_auth.c

```diff
@@ -1508,7 +1509,15 @@ exi_cache_trim(struct exportinfo *exi)
 		 * used for NFSAUTH_CACHE_TRIM seconds.
 		 */
 		for (c = avl_first(tree); c != NULL; c = AVL_NEXT(tree, c)) {
-			rw_enter(&c->authc_lock, RW_WRITER);
+			/*
+			 * We are being called by the kmem subsystem to reclaim
+			 * memory so don't block if we can't get the lock.
+			 */
+			if (rw_tryenter(&exi->exi_cache_lock, RW_WRITER) == 0) {
```

Comment:
WHOA!  Do you want the exi_cache_lock here?  Or authc lock?  Also, if its the exi_cache_lock, shouldn't this tryenter logic be up on line 1505-ish?